### PR TITLE
fix(core): make start_time/end_time inclusive on GET /logs

### LINKED ISSUE #8660: feature request: API endpoint /api/logs to support filtering by time.
### What problem did you meet?

Audit logs API endpoint /api/logs only supports pagination but not filtering by time.

### Describe what you'd like Logto to have

I would love to see this endpoints to support from/to arguments that work inclusively. Either via ISO timestamps or  Unix epoch milliseconds.

From/to arguments should be optional - not passing them will keep the current behavior of the endpoints.

When the from/to arguments are passed we're going to filter the logs based on the default page size and control the amount of entries with page_size.

### PR DESCRIPTION
**Pull request by Kathircpe**
closes #8660 

## Summary

Changed > to >= and < to <= on createdAt filters in GET /api/logs, making start_time/end_time inclusive, with relaxed validation (start > end instead of start >= end) and updated tests.

Linked issue can be closed. The issue requests for a feature and the feature exists right now, so I just made a little changes.

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

**Review by charIeszhao — APPROVED** [review]

File: packages/core/src/queries/log.ts [hunk 1]
```diff
@@ -65,10 +65,10 @@ const buildLogConditionSql = (logCondition: LogCondition) =>
       // falsy and would silently drop the filter.
       startTime === undefined
         ? sql``
-        : sql`${fields.createdAt} > to_timestamp(${startTime}::double precision / 1000)`,
+        : sql`${fields.createdAt} >= to_timestamp(${startTime}::double precision / 1000)`,
```
Comment (wangsijie) [inline packages/core/src/queries/log.ts:73]: API is now inclusive, but Console still applies exclusive ±1ms biases (see `resolveTimeWindow` in `packages/console/src/components/AuditLogTable/components/TimeRangePicker/preset.ts` and the `startTime + 1` undo). Adjacent-day edges can over-include by 1ms; comments/tests are also stale.

Please update Console to true inclusive day bounds, or track a follow-up so core/console don't ship mismatched contracts.

File: packages/core/src/routes/log.openapi.json [hunk 2]
```diff
@@ -29,12 +29,12 @@
           {
             "name": "start_time",
             "in": "query",
-            "description": "Exclusive lower bound on `createdAt`, in unix milliseconds. When set, returns only logs where `createdAt > start_time`. Recommended for clients paginating across very large log ranges to keep query latency consistent."
+            "description": "Inclusive lower bound on `createdAt`, in unix milliseconds. When set, returns only logs where `createdAt >= start_time`."
```
Comment (wangsijie) [inline packages/core/src/routes/log.openapi.json]: Unrelated drop: the previous `start_time` note about using time windows for pagination on large log ranges was removed. Please restore or rephrase it.

**Review by charIeszhao — APPROVED** [review]

**Review by wangsijie — COMMENTED** [review]
Found one precision edge case during local end-to-end verification.

File: packages/core/src/queries/log.ts [hunk 3]
```diff
@@ -65,10 +65,10 @@ const buildLogConditionSql = (logCondition: LogCondition) =>
...
       endTime === undefined
         ? sql``
-        : sql`${fields.createdAt} < to_timestamp(${endTime}::double precision / 1000)`,
+        : sql`${fields.createdAt} <= to_timestamp(${endTime}::double precision / 1000)`,
```
Comment (wangsijie) [inline packages/core/src/queries/log.ts]: PostgreSQL timestamps retain microsecond precision, while `createdAt` is exposed by the API as Unix milliseconds. This means a row stored at `...59.999500` is serialized as the same millisecond as `end_time=...999`, but is excluded by this `<=` comparison.

I reproduced this locally: a row stored at epoch `1784217599999.500` is returned as `createdAt: 1784217599999`, yet it is missing when querying with `end_time=1784217599999`. This also causes the Console’s `endOfDay(...).999` filter to miss logs from the final 0.999ms of the selected day.

Could we treat the upper bound as an inclusive millisecond bucket, for example:

```sql
created_at < to_timestamp(end_time / 1000) + interval '1 millisecond'
```

It would also be useful to add a regression test with a timestamp such as `...59.999500`.

Comment (Kathircpe) [reply packages/core/src/queries/log.ts]: I  have made the requested changes for the extreme edge case.

**Comment by Kathircpe** [comment]
Hey @wangsijie , I have addressed your comments.
It is ready for review.

**Review by wangsijie — APPROVED** [review]

**Comment by simeng-li** [comment]
@Kathircpe please check the failed integration tests

**Comment by Kathircpe** [comment]
@simeng-li, I've fixed the test failure.

**Comment by simeng-li** [comment]
> @simeng-li, I've fixed the test failure.

Blocked by the missing signature in the commits.

**Comment by Kathircpe** [comment]
@simeng-li Done. Thanks for finding the failure cause for me.
